backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/FilterPrivilegedDatasetsHandler.java

```diff
@@ -98,50 +100,49 @@ public FilterPrivilegedDatasetsHandler(Executor executor, FutureJdbi jdbi) {
 
               // If all dataset with linked dataset versions found in database then checking
               // authorization for the current user for those datasets
-              List<InternalFuture<String>> internalFutures = new LinkedList<>();
+              List<InternalFuture<Optional<String>>> internalFutures = new LinkedList<>();
               for (String datasetId : datasetIds) {
-                try {
-                  internalFutures.add(
-                      InternalFuture.completedInternalFuture(datasetId)
-                          .thenCompose(
-                              id ->
-                                  permissionCheck
-                                      .getEntityPermissionBasedOnResourceTypes(
-                                          Collections.singletonList(datasetId),
-                                          ModelDBActionEnum.ModelDBServiceActions.READ,
-                                          ModelDBResourceEnum.ModelDBServiceResourceTypes.DATASET)
-                                      .thenAccept(
-                                          allowed -> {
-                                            if (!allowed) {
-                                              throw new PermissionDeniedException(
-                                                  "Permission denied");
-                                            }
-                                          },
-                                          executor)
-                                      .thenCompose(
-                                          unused -> InternalFuture.completedInternalFuture(id),
-                                          executor),
-                              executor));
-
-                } catch (PermissionDeniedException ex) {
-                  LOGGER.trace(ex.getMessage());
-                  // We are throws error while creating entities
-                  // if we are fetching entities then we are just skip those datasets from response
-                  // and skipped PermissionDeniedException on fetch
-                  if (errorOut) {
-                    return InternalFuture.failedStage(ex);
-                  }
-                }
```

Comment:
error out using try catch is not worked with InternalFuture stuff so improve the logic